core/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -59,16 +64,40 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
         final RestChannelConsumer action = prepareRequest(request, client);
 
         // validate unconsumed params, but we must exclude params used to format the response
-        final List<String> unconsumedParams =
-            request.unconsumedParams().stream().filter(p -> !responseParams().contains(p)).collect(Collectors.toList());
+        // use a sorted set so the unconsumed parameters appear in a reliable sorted order
+        final SortedSet<String> unconsumedParams =
+            request.unconsumedParams().stream().filter(p -> !responseParams().contains(p)).collect(Collectors.toCollection(TreeSet::new));
 
         // validate the non-response params
         if (!unconsumedParams.isEmpty()) {
-            final String message = String.format(
-                Locale.ROOT,
-                "request [%s] contains unrecognized parameters: %s",
-                request.path(),
-                unconsumedParams.toString());
+            String message = "request [" + request.path() + "] contains unrecognized parameters: ";
+            boolean first = true;
+            for (final String unconsumedParam : unconsumedParams) {
+                final LevensteinDistance ld = new LevensteinDistance();
+                final List<Tuple<Float, String>> scoredParams = new ArrayList<>();
+                for (String consumedParam : request.consumedParams()) {
+                    final float distance = ld.getDistance(unconsumedParam, consumedParam);
+                    if (distance > 0.5f) {
```

Comment:
Also, as far as using the same code, I really don't want to try to make an abstraction that captures both uses, they are slightly different. Abstracting them is possible, but it just makes it messy and we only have two uses to base the abstraction on. If we ever hit a third, we can maybe consider it. For now, I strongly prefer keeping it simple and as-is.